### PR TITLE
Introduce per test suite timeout

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area testing dev-productivity
/kind bug
/priority normal

**What this PR does / why we need it**:

This PR introduces a general per test suite timeout for unit tests, when running via `make test-cover`, e.g. in the CI verify step.

Unit tests should be quick / fail early. Long running unit tests indicate either some issues in the tests or some performance problem in the underlying production code itself.

**Which issue(s) this PR fixes**:
I had some tests, that took very long to fail, which was annoying to notice/debug.

**Special notes for your reviewer**:

I didn't add a timeout to `make test`, i.e. for manually/locally executing unit tests.
The reason for that is, that the go test `-timeout` flag is not "cacheable" and therefore will cause all test suites to be executed everytime, although the results may be cached ([ref](https://golang.org/cmd/go/#hdr-Testing_flags)).

**Release note**:

```other developer
Unit tests are now limited to a timeout of 2 minutes per test suite.
```

### CODE REVIEW

**Review by rfranzke — DISMISSED** [review]
/lgtm

File: pkg/utils/secrets/basic_auth_test.go [hunk 1]
```diff
@@ -81,7 +81,8 @@ var _ = Describe("Basic Auth Secrets", func() {
 				Expect(err).NotTo(HaveOccurred())
 				compareCurrentAndExpectedBasicAuth(obj, expectedBasicAuthObject, false, false)
 			})
-			It("should properly generate Basic Auth Object with Bcrypt hashed password if specified in config", func() {
+			// pending because test is consistently slow, see https://github.com/gardener/gardener/issues/3362
+			PIt("should properly generate Basic Auth Object with Bcrypt hashed password if specified in config", func() {
```
Comment (rfranzke) [inline pkg/utils/secrets/basic_auth_test.go]: Now that #3365 was opened you can revert this change

Comment (timebertt) [reply pkg/utils/secrets/basic_auth_test.go]: Sure, done.

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Review by rfranzke — APPROVED** [review]
